### PR TITLE
Feat: Add Actor Credentials

### PR DESCRIPTION
**Pull request by FirestarJes**

### CODE REVIEW

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Repositories/ActorErrorHandler.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+using System.ComponentModel.DataAnnotations;
+
+namespace Energinet.DataHub.MarketParticipant.Domain.Repositories;
+
+public static class ActorErrorHandler
+{
+    public static System.Exception HandleActorError(ActorError source) => source switch
+    {
+        ActorError.ThumbprintCredentialsConflict => new ValidationException("An actor with the same certificate thumbprint already exists"),
+        ActorError.ClientSecretCredentialsConflict => new ValidationException("An actor with the same Client secret identifier already exists"),
```
Comment (defectiveAi) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Repositories/ActorErrorHandler.cs:25]: ClientSecretCredentialsConflict is not a domain/validation error, as the secret is always created by infrastructure. If this conflict happens, we have a bug.

In the other case, it is a validation error, as a user can upload the same cert twice for two different actors.

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Infrastructure/Persistence/Repositories/ActorRepository.cs [hunk 2]
```diff
@@ -54,8 +55,24 @@ public async Task<ActorId> AddOrUpdateAsync(Actor actor)
...
+        {
+            await _marketParticipantDbContext.SaveChangesAsync().ConfigureAwait(false);
+        }
+        catch (DbUpdateException ex) when (
+            ex.InnerException is SqlException inner &&
+            inner.Message.Contains("UQ_ActorCertificateCredentials_Thumbprint", StringComparison.InvariantCultureIgnoreCase))
+        {
+            return new(ActorError.ThumbprintCredentialsConflict);
+        }
+        catch (DbUpdateException ex) when (
```
Comment (defectiveAi) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.Infrastructure/Persistence/Repositories/ActorRepository.cs:68]: Do not catch this one, see previous comment.

**Review by defectiveAi — CHANGES_REQUESTED** [review]

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Infrastructure/Persistence/Repositories/ActorRepository.cs (hunk 2 above)
Comment (FirestarJes) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.Infrastructure/Persistence/Repositories/ActorRepository.cs:68]: Fixed

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Repositories/ActorErrorHandler.cs (hunk 1 above)
Comment (FirestarJes) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Repositories/ActorErrorHandler.cs:25]: Fixed

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Actor.cs [hunk 3]
```diff
@@ -113,6 +131,11 @@ public ActorStatus Status
     /// </summary>
     public ActorName Name { get; set; }
 
+    /// <summary>
+    /// The list of credentials for the current actor.
+    /// </summary>
+    public ICollection<ActorCredentials> Credentials { get; }
```
[the remark this answers is not in the document]
Comment (FirestarJes) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Actor.cs:137]: Fixed

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Actor.cs [hunk 4]
```diff
@@ -39,14 +40,29 @@ public sealed class Actor : IPublishDomainEvents
         _actorStatusTransitioner = new ActorStatusTransitioner();
     }
 
+    public Actor(
+        OrganizationId organizationId,
+        ActorNumber actorNumber,
+        ActorName actorName,
+        ActorCredentials credentials)
```
Comment (defectiveAi) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Actor.cs:47]: Remove credentials from ctor. We do not have credentials at construction time.

Comment (FirestarJes) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Actor.cs]: Fixed

**Review by defectiveAi — APPROVED** [review]

**Pull request merged**